gprofiler/profilers/python.py

```diff
@@ -89,46 +89,59 @@ def _add_versions_to_stacks(
 class PythonMetadata(ApplicationMetadata):
     _PYTHON_VERSION_TIMEOUT = 3
 
-    def _get_python_version(self, process: Process) -> str:
-        if not os.path.basename(process.exe()).startswith("python"):
-            # TODO: for dynamic executables, find the python binary that works with the loaded libpython, and
-            # check it instead. For static executables embedding libpython - :shrug:
-            raise NotImplementedError
-
-        python_path = f"/proc/{get_process_nspid(process.pid)}/exe"
-
-        def _run_python_version() -> "CompletedProcess[bytes]":
-            return run_process(
-                [
-                    python_path,
-                    "-V",
-                ],
-                stop_event=self._stop_event,
-                timeout=self._PYTHON_VERSION_TIMEOUT,
-            )
+    def _run_process_python_python(self, process: Process, args: List[str]) -> Optional[str]:
+        try:
+            if not os.path.basename(process.exe()).startswith("python"):
+                # TODO: for dynamic executables, find the python binary that works with the loaded libpython, and
+                # check it instead. For static executables embedding libpython - :shrug:
+                raise NotImplementedError
+
+            python_path = f"/proc/{get_process_nspid(process.pid)}/exe"
+
+            def _run_python_version() -> "CompletedProcess[bytes]":
+                return run_process(
+                    [
+                        python_path,
+                    ]
+                    + args,
+                    stop_event=self._stop_event,
+                    timeout=self._PYTHON_VERSION_TIMEOUT,
+                )
 
-        cp = run_in_ns(["pid", "mnt"], _run_python_version, process.pid)
-        version = cp.stdout.decode().strip()
-        if version:
-            return version
+            cp = run_in_ns(["pid", "mnt"], _run_python_version, process.pid)
+            output = cp.stdout.decode().strip()
+            if not output and args == ["-V"]:
```

Comment:
Very specific...
Return `stdout, stderr`, and have one more function:
```python
def get_python_version(self, process):
    self. _run_process_python_python(process, ['-V'])
    if python2:

version = self.get_python_version()
sys_maxunicode, _ = self. _run_process_python_python(process, ['-S', '-c', ..])
```

Reply:
Reorganized